pkg/hostagent/opflex.go

```diff
@@ -215,7 +214,10 @@ func (agent *HostAgent) updateOpflexConfig() {
 
 	newNodeConfig := agent.discoverHostConfig()
 	if newNodeConfig == nil {
-		panic(errors.New("Node configuration autodiscovery failed"))
+                // FIXME
```

Comment:
Per the "fixme" here, should we not have conditional logic here to only proceed if its "overlay" mode?